masonry_core/src/core/action.rs

```diff
@@ -24,10 +29,40 @@ pub enum Action {
     CheckboxToggled(bool),
     // FIXME - This is a huge hack
     /// Other.
-    Other(Box<dyn Any + Send>),
+    Other(Box<dyn DebugAndAny + Send>),
+}
+
+/// Empty type.
+///
+/// Used by widgets which never emit actions.
+///
+/// Will be replaced with `!` once stable.
+pub enum NoAction {}
+
+/// Trait alias for traits [`Debug`] and [`Any`].
+///
+/// Used to be able to declare `Box<dyn DebugAndAny>`.
+pub trait DebugAndAny: Debug + Any {}
```

Comment:
I'd consider calling this `AnyAction`. That opens the door to future trait bounds without needing to rename all uses of this.

(That name suggestion is based on this being exactly the same idea as https://github.com/linebender/xilem/blob/825cfd88ebba3c7c838921a9755fe233c19eec50/xilem_core/src/message.rs#L44-L123; you may as well import similar tests and utility methods as well)

Reply:
Yeah, the point where I saw AnyMessage was when I decided to just commit what I had and stop.

My plan is to create a clean slate PR that moves AnyMessage to Masonry, and when it's merged, rebase this PR on top of it.